### ISSUE TITLE
ERESOLVE errors within EAS build for native-base and Expo

### ISSUE THREAD

**Issue by MehmetKaplan** [GeekyAnts/NativeBase#4647]
### Description

Expo 44 and native-base (version `^3.0.0`, or more precisely `native-base@3.3.6` can not be built with EAS which uses `npm` to install the dependencies.

### CodeSandbox/Snack link

This is a problem for building packages so please check **"Steps to produce"** section.

### Steps to reproduce

1. Generate a blank expo project:
`expo init test-native-base-peer-dependency`
2. Go into the new directory:
`cd test-native-base-peer-dependency`
3. Modify `package.json` file, with below content:
```json
{
        "name": "test-native-base-peer-dependency",
        "version": "1.0.0",
        "main": "node_modules/expo/AppEntry.js",
        "scripts": {
                "start": "expo start",
                "android": "expo start --android",
                "ios": "expo start --ios",
                "web": "expo start --web",
                "eject": "expo eject"
        },
        "dependencies": {
                "@expo/samples": "2.1.1",
                "@expo/vector-icons": "^12.0.0",
                "@react-native-async-storage/async-storage": "~1.15.0",
                "@react-native-community/datetimepicker": "4.0.0",
                "@react-native-community/netinfo": "7.1.3",
                "@react-native-picker/picker": "2.2.1",
                "expo": "^44.0.0",
                "expo-ads-admob": "~12.0.0",
                "expo-app-loading": "~1.3.0",
                "expo-apple-authentication": "~4.1.0",
                "expo-application": "~4.0.1",
                "expo-asset": "~8.4.5",
                "expo-auth-session": "~3.5.0",
                "expo-av": "~10.2.0",
                "expo-facebook": "~12.1.0",
                "expo-font": "~10.0.4",
                "expo-haptics": "~11.1.0",
                "expo-image-manipulator": "~10.2.0",
                "expo-random": "~12.1.1",
                "expo-screen-orientation": "~4.1.1",
                "expo-sharing": "~10.1.0",
                "expo-tracking-transparency": "~2.1.0",
                "moment": "^2.24.0",
                "native-base": "^3.0.0",
                "react": "17.0.1",
                "react-native": "0.64.3",
                "react-native-gesture-handler": "~2.1.0",
                "react-native-modal-datetime-picker": "^8.6.0",
                "react-native-picker-select": "^8.0.0",
                "react-native-progress": "^4.1.2",
                "react-native-progress-circle": "^2.1.0",
                "react-native-safe-area-context": "3.3.2",
                "react-native-svg": "12.1.1",
                "react-native-view-shot": "3.1.2",
                "react-native-webview": "11.15.0",
                "react-redux": "^6.0.0",
                "redux": "^4.0.1"
        },
        "devDependencies": {
                "babel-preset-expo": "9.0.1"
        },
        "resolutions": {},
        "private": true
}
```
4. Install the packages.
`npm install`
5. See error
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: test-native-base-peer-dependency@1.0.0
npm ERR! Found: react@17.0.1
npm ERR! node_modules/react
npm ERR!   react@"17.0.1" from the root project
npm ERR!   peer react@"*" from native-base@3.3.6
npm ERR!   node_modules/native-base
npm ERR!     native-base@"^3.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"17.0.2" from react-dom@17.0.2
npm ERR! node_modules/react-dom
npm ERR!   peer react-dom@"*" from native-base@3.3.6
npm ERR!   node_modules/native-base
npm ERR!     native-base@"^3.0.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

### NativeBase Version

3.3.6

### Platform

- [ ] Android
- [ ] CRA
- [X] Expo
- [ ] iOS
- [ ] Next

### Additional Information

The Expo eas build also uses `npm`. That's why this problem prevents deploying codes to the production as well.

**Comment by Viraj-10** [comment]
HI @MehmetKaplan, peer dependency issues are resolved in the latest version of native-base.
Please try using the following pakage.json with correct version.
PS :- `react-native-modal-datetime-picker` giving peer dependency error.
### Pakage.json
```
{
  "name": "my-app-js",
  "version": "1.0.0",
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web",
    "eject": "expo eject"
  },
  "dependencies": {
    "@expo/samples": "^3.0.3",
    "@expo/vector-icons": "^12.0.5",
    "@react-native-async-storage/async-storage": "^1.16.1",
    "@react-native-community/datetimepicker": "^5.1.0",
    "@react-native-community/netinfo": "^8.0.0",
    "@react-native-picker/picker": "^2.2.1",
    "expo": "~44.0.0",
    "expo-ads-admob": "^12.0.1",
    "expo-app-loading": "^1.3.0",
    "expo-apple-authentication": "^4.1.0",
    "expo-application": "^4.0.2",
    "expo-asset": "^8.4.6",
    "expo-auth-session": "^3.5.0",
    "expo-av": "^10.2.1",
    "expo-facebook": "^12.1.1",
    "expo-font": "^10.0.5",
    "expo-haptics": "^11.1.1",
    "expo-image-manipulator": "^10.2.1",
    "expo-random": "^12.1.2",
    "expo-screen-orientation": "^4.1.2",
    "expo-sharing": "^10.1.1",
    "expo-status-bar": "~1.2.0",
    "expo-tracking-transparency": "^2.1.0",
    "moment": "^2.29.1",
    "native-base": "^3.3.7",
    "react": "17.0.1",
    "react-dom": "17.0.1",
    "react-native": "0.64.3",
    "react-native-gesture-handler": "^2.2.0",
    "react-native-picker-select": "^8.0.4",
    "react-native-progress": "^5.0.0",
    "react-native-progress-circle": "^2.1.0",
    "react-native-safe-area-context": "3.1.9",
    "react-native-svg": "12.1.0",
    "react-native-view-shot": "^3.1.2",
    "react-native-web": "0.17.1",
    "react-native-webview": "^11.17.2",
    "react-redux": "^7.2.6",
    "redux": "^4.1.2"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9"
  },
  "private": true
}

```

**Comment by MehmetKaplan** [comment]
Hi Viraj,

Thanks for the answer. Using it as a clue, my conclusion is; native-base searches `"react-dom": "17.0.2"` and it in turn looks for `"react": "17.0.2"` which conflicts with expo's `"react": "17.0.1"`.

So adding `"react-dom": "17.0.1"` as dependency solved the problem. (Please let me know if this conclusion is wrong.)

I think this can be closed. Thank you for the help.

**Comment by jukialen** [comment]
I have the same problem, but other communication:
`Found: react-native-safe-area-context@3.3.2
npm ERR! node_modules/react-native-safe-area-context
npm ERR!   react-native-safe-area-context@"3.3.2" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react-native-safe-area-context@"3.1.9" from native-base@3.3.7
npm ERR! node_modules/native-base
npm ERR!   native-base@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.`

I used npm update but this didn't help.
I created project with template tabs (Expo).
